### PR TITLE
✨(frontend) Allow download of training contracts

### PR DESCRIPTION
**Pull request by rlecellier**
## Purpose

joanie is now able to generate contracts archive for given organization
and courseProductRelation ids.
Here we implemente this feature on the course details pages

### CODE REVIEW

File: src/frontend/js/pages/TeacherDashboardContractsLayout/hooks/useDownloadContractArchive/contractArchiveLocalStorage.spec.ts
[the remark this answers is not in the document]
Comment (rlecellier) [reply src/frontend/js/pages/TeacherDashboardContractsLayout/hooks/useDownloadContractArchive/contractArchiveLocalStorage.spec.ts:1]: It's not the same after each.
first time we reset `localStorageArchiveFilters` and second time `localStorageExistingArchiveFilters`

It's not easy to read but everycase is tested, doing the same without each would make a crazy long test file that would not be easy to read either.

[the remark this answers is not in the document]
Comment (rlecellier) [reply src/frontend/js/pages/TeacherDashboardContractsLayout/hooks/useDownloadContractArchive/contractArchiveLocalStorage.spec.ts:1]: TODO: merge both each.

File: src/frontend/js/pages/TeacherDashboardContractsLayout/components/ContractActionsBar/index.tsx [hunk 1]
```diff
@@ -38,7 +37,12 @@ const ContractActionsBar = ({ organizationId, courseProductRelationId }: Contrac
             />
           </div>
         )}
-        {canDownloadContracts && <BulkDownloadContractButton organizationId={organizationId} />}
+        {hasContractToDownload && (
```
[the remark this answers is not in the document]
Comment (rlecellier) [reply src/frontend/js/pages/TeacherDashboardContractsLayout/components/ContractActionsBar/index.tsx:40]: slack resume.

I didn't remove a ability check but remove this variable: 
```ts
const canDownloadContracts = hasContractToDownload && !courseProductRelationId
```
It's no longer needed to check `courseProductRelationId`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,9 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Fixed
 
+- Downloaded contract archive on product page was broken, it now work
+  propertly by sending the course_product_relation_id in the archive 
+  creation request.
```
Comment (jbpenrath) [inline CHANGELOG.md:20]:
```suggestion
- Downloaded contract archive on product page was broken, it now works
  properly by sending the course_product_relation_id in the archive 
  creation request.
```

**Review by jbpenrath — APPROVED** [review]

**Pull request merged**